src/main/java/uk/gov/hmcts/reform/managecase/api/errorhandling/RestExceptionHandler.java

```diff
@@ -0,0 +1,64 @@
+package uk.gov.hmcts.reform.managecase.api.errorhandling;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+import javax.validation.ConstraintViolationException;
+import javax.validation.ValidationException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR;
+
+@RestControllerAdvice
+public class RestExceptionHandler extends ResponseEntityExceptionHandler {
```

Comment:
exceptions not logged. could you please log exceptions?
Also, so lines are just repeated. It would be better to have a common method to translate and ApiError object into an ResponseEntity.

Reply:
common method done. logging 500 errors already. Do we need to log 400 as well? if so what's the logging level (warn?) ?

Reply:
Thank you. I would recommend HttpStatus.FORBIDDEN to be logged at WARN level, as that is something we normally wouldn't expect and need to investigate should it ever happen. Other 4XX can just be logged at debug level.